File: app/src/main/java/com/example/myapplication2/MainActivity.java
package com.example.myapplication2;

import android.content.Intent;
import android.content.res.TypedArray;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.example.myapplication2.databinding.ActivityMainBinding;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private RecyclerView rvtense;
    private final ArrayList<tense> list = new ArrayList<>();
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        rvtense = findViewById(R.id.materi);
        rvtense.setHasFixedSize(true);

        list.addAll(getListtense());
        showRecyclerList();

        setSupportActionBar(binding.toolbar);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setHomeButtonEnabled(true);
        }
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }

    private ArrayList<tense> getListtense() {
        String[] dataName = getResources().getStringArray(R.array.data_name);
        TypedArray dataFoto = getResources().obtainTypedArray(R.array.data_foto);
        String[] dataMateri = getResources().getStringArray(R.array.tenses);
        String[] dataContoh = getResources().getStringArray(R.array.contoh);

        ArrayList<tense> listtense = new ArrayList<>();
        for (int i = 0; i < dataName.length; i++) {
            tense tense = new tense(dataName[i], dataFoto.getResourceId(i, -1), dataMateri[i], dataContoh[i]);
            listtense.add(tense);
        }
        dataFoto.recycle();
        return listtense;
    }

    private void showRecyclerList() {
        rvtense.setLayoutManager(new LinearLayoutManager(this));
        listAdapter listHeroAdapter = new listAdapter(list);
        rvtense.setAdapter(listHeroAdapter);
        listHeroAdapter.setOnItemClickCallback(this::showSelectedHero);
    }

    private void showSelectedHero(tense hero) {
        Intent intent;
        if (hero.getName().equals("Metamorfosis Sempurna")) {
            intent = new Intent(MainActivity.this, Materi.class);
        } else if (hero.getName().equals("Metamorfosis Tidak Sempurna")) {
            intent = new Intent(MainActivity.this, Materi2.class);
        } else {
            intent = new Intent(MainActivity.this, Materi3.class);
        }
        intent.putExtra(Materi.EXTRA_MATERI, hero);
        intent.putExtra(Materi.EXTRA_NAME, hero.getName());
        startActivity(intent);
    }
}

File: app/src/main/java/com/example/myapplication2/MainActivity.kt
package com.example.myapplication2

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.myapplication2.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var rvTense: RecyclerView
    private val list = ArrayList<tense>()
    private lateinit var binding : ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        rvTense = findViewById(R.id.materi)
        rvTense.setHasFixedSize(true)

        list.addAll(getListTense)
        showRecyclerList()

        supportActionBar?.title = "Materi"

        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setLogo(R.drawable.ic_back)
        supportActionBar?.setDisplayUseLogoEnabled(true)
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return super.onSupportNavigateUp()
    }

    private val getListTense: ArrayList<tense>
        get() {
            val dataName = resources.getStringArray(R.array.data_name)
            val datafoto = resources.obtainTypedArray(R.array.data_foto)

            val dataMateri = resources.getStringArray(R.array.tenses)
            val dataContoh = resources.getStringArray(R.array.contoh)
            val listTense = ArrayList<tense>()
            for (i in dataName.indices) {
                val tense =
                    tense(dataName[i], datafoto.getResourceId(i, -1), dataMateri[i], dataContoh[i])

                listTense.add(tense)
            }
            return listTense
        }


    private fun showRecyclerList() {
        rvTense.layoutManager = LinearLayoutManager(this)
        val listHeroAdapter = listAdapter(list)
        rvTense.adapter = listHeroAdapter
        listHeroAdapter.setOnItemClickCallback(object : listAdapter.OnItemClickCallback {
            override fun onItemClicked(data: tense) {
                showSelectedHero(data)
            }
        })
    }

    private fun showSelectedHero(hero: tense) {
        val intent = Intent(this@MainActivity, Materi::class.java)
        intent.putExtra(Materi.EXTRA_MATERI, hero)
        intent.putExtra(Materi.EXTRA_NAME, hero.name)
        startActivity(intent)
    }

}